### ISSUE TITLE
[FR] Animate .hide() & .show() like .open() & .close()

### ISSUE THREAD

**Issue by paricleu** [akshathjain/sliding_up_panel#164]
Right now the hide and show methods aren't animated. It would be nice to be able to hide the panel completely to highlight the widget behind the panel.

**Comment by kkizlaitis** [comment]
Yes, this would be a highly appreciated feature! :) Right now, when you call `hide()`, the panel animates to the `minHeight` and then disappears instantly. It should animate to zero height instead.

**Comment by kar1981** [comment]
Yes this is really needed at least as an option to animate it. I have a use case where I call .hide() from a button to dismiss the panel in collapsed state and now it abruptly disappears instead of animating sliding down. I could have use the snap point feature but there is no way to lockout the sliding when the panel is at snap point. 

In google maps the behaviour is demonstrated by:
1. Select a poi on the map such as a restaurant
2. Slide down the panel to the most collapsed state (only shows place name and rating info)
At this point user cannot further slide it down to close the panel completely from view.
3. Tap a "blank spot" on the map and the panel will animate to close completely.

As it stands now, this behaviour can't be replicated using this widget.

**Comment by stjfk** [comment]
+1
